lib/datadog/grok/src/grok_filter.rs

```diff
@@ -153,5 +158,123 @@ pub fn apply_filter(value: &Value, filter: &GrokFilter) -> Result<Value, GrokRun
                 value.to_string(),
             )),
         },
+        GrokFilter::Date(format, tz_regex_opt, target_tz, with_tz) => match value {
```

Comment:
This `apply_filter` function is getting quite big.

I think it would be worth splitting each branch in the `match` into a separate function. That is a subjective thing, and I would be interested in what other people felt, but it would save the big wall of code that this is turning into..

Reply:
yep, agree, will split it.